const-oid/oiddbgen/src/kind.rs

```diff
@@ -0,0 +1,59 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::quote;
+
+use crate::spec::Spec;
+
+use std::{
+    collections::BTreeMap,
+    ops::{Deref, DerefMut},
+};
+
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub struct Kind(BTreeMap<Ident, Spec>);
+
+impl Deref for Kind {
+    type Target = BTreeMap<Ident, Spec>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for Kind {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl Kind {
+    pub fn symbols(&self, path: TokenStream) -> TokenStream {
+        let mut stream = TokenStream::default();
+
+        for (spec, s) in &self.0 {
+            stream.extend(s.symbols(quote! { #path::#spec }))
+        }
+
+        stream
+    }
+
+    pub fn module<'a>(&'a self, kind: &Ident, docs: &TokenStream) -> TokenStream {
```

Comment:
I think it's generally considered non-idiomatic to have non-static methods in addition to impl `Deref`/`DerefMut`.

Reply:
Fixed.